internal/alertmanager/currentalert/process.go

```diff
@@ -238,8 +238,56 @@ func addPvc(log logr.Logger, alertLabels model.LabelSet, alertAnnotations model.
 	return nil
 }
 
-func resizePvc(labels model.LabelSet, annotiations model.LabelSet, client client.Client) error {
-	//TODO
+func resizePvc(log logr.Logger, labels model.LabelSet, annotiations model.LabelSet, client client.Client) error {
+	cr, err := k8sutil.GetCr(string(labels["kafka_cr"]), string(labels["namespace"]), client)
+	if err != nil {
+		return err
+	}
+
+	pvc, err := getPvc(string(labels["persistentvolumeclaim"]), string(labels["namespace"]), client)
+	if err != nil {
+		return err
+	}
+
+	for i, broker := range cr.Spec.Brokers {
+		if strconv.Itoa(int(broker.Id)) == pvc.Labels["brokerId"] {
+			var storageConfigs []v1beta1.StorageConfig
+
+			if broker.BrokerConfig == nil {
+				storageConfigs = cr.Spec.BrokerConfigGroups[cr.Spec.Brokers[i].BrokerConfigGroup].StorageConfigs
+			} else {
+				storageConfigs = broker.BrokerConfig.StorageConfigs
+			}
+
+			for i, c := range storageConfigs {
+				if c.MountPath == pvc.Annotations["mountPath"] {
+					size := *c.PvcSpec.Resources.Requests.Storage()
+
+					size.Add(resource.MustParse(string(annotiations["diskSize"])))
```

Comment:
Extend `resizePvcValidator` or implement a new one  to validate that there is a `diskSize` annotation on the alert and is in the correct format. We should not use `MustParse` as the can exit with `panic` but rather ensure that the execution reaches this part only if the received alert is valid for processing.

The `diskSize` is misleading as in fact it's an increment, thus `diskSizeIncrement` or `incrementBy` would be a better name.

Reply:
I changed it to `incrementBy` and I implemented some basic checking logic in `resizePvcValidator`. 

I will say you guys should consider opening a ticket to do the same thing to `addPvc` as well as `upScale` they do not do any validation on `diskSize` in the `*Validator` and they all implement `resources.MustParse`. I basically followed `addPvc` as close as I could to not change the style of the code or make it look too different.